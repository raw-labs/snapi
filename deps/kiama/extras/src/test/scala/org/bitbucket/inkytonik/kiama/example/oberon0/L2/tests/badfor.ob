MODULE badfor;

CONST
    aconst = 10;
    
TYPE
    atype = INTEGER;    

VAR
    avar : INTEGER;
    bvar : BOOLEAN;
    cvar : INTEGER;

BEGIN
    FOR aconst := 1 TO 10 DO
        avar := 1
    END;

    FOR atype := 1 TO 10 DO
        avar := 1
    END;

    FOR bvar := FALSE TO TRUE DO
        avar := 42
    END;

    FOR avar := 1 TO 2 BY cvar * 2 DO
        avar := 42
    END;

    FOR dvar := 1 TO 2 DO
        dvar := 99
    END;

    FOR avar := 8 TO 10 BY 3 DIV 0 DO
        cvar := 100
    END
END badfor.
