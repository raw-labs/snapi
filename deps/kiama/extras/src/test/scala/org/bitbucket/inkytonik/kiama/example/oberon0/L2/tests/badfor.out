* _pp.ob
MODULE badfor;

    CONST
        aconst = 10;

    TYPE
        atype = INTEGER;

    VAR
        avar : INTEGER;
        bvar : BOOLEAN;
        cvar : INTEGER;

BEGIN
    FOR aconst := 1 TO 10 DO
        avar := 1
    END;
    FOR atype := 1 TO 10 DO
        avar := 1
    END;
    FOR bvar := FALSE TO TRUE DO
        avar := 42
    END;
    FOR avar := 1 TO 2 BY cvar * 2 DO
        avar := 42
    END;
    FOR dvar := 1 TO 2 DO
        dvar := 99
    END;
    FOR avar := 8 TO 10 BY 3 DIV 0 DO
        cvar := 100
    END
END badfor.
[2]* errors
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L2/tests/badfor.ob:15:9:error: illegal FOR loop control variable aconst
[3]    FOR aconst := 1 TO 10 DO
[3]        ^
[2]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L2/tests/badfor.ob:19:9:error: atype cannot be used in an expression
[2]    FOR atype := 1 TO 10 DO
[2]        ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L2/tests/badfor.ob:19:9:error: illegal FOR loop control variable atype
[3]    FOR atype := 1 TO 10 DO
[3]        ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L2/tests/badfor.ob:23:9:error: got BOOLEAN, but expected INTEGER
[3]    FOR bvar := FALSE TO TRUE DO
[3]        ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L2/tests/badfor.ob:23:17:error: got BOOLEAN, but expected INTEGER
[3]    FOR bvar := FALSE TO TRUE DO
[3]                ^
[3]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L2/tests/badfor.ob:23:26:error: got BOOLEAN, but expected INTEGER
[3]    FOR bvar := FALSE TO TRUE DO
[3]                         ^
[2]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L2/tests/badfor.ob:27:27:error: expression is not constant
[2]    FOR avar := 1 TO 2 BY cvar * 2 DO
[2]                          ^
[2]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L2/tests/badfor.ob:31:9:error: dvar is not declared
[2]    FOR dvar := 1 TO 2 DO
[2]        ^
[2]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L2/tests/badfor.ob:32:9:error: dvar is not declared
[2]        dvar := 99
[2]        ^
[2]src/test/scala/org/bitbucket/inkytonik/kiama/example/oberon0/L2/tests/badfor.ob:35:34:error: division by zero in constant expression
[2]    FOR avar := 8 TO 10 BY 3 DIV 0 DO
[2]                                 ^
